text_extensions_for_pandas/io/watson.py

```diff
@@ -0,0 +1,136 @@
+#
+#  Copyright (c) 2020 IBM Corp.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+################################################################################
+# watson.py
+#
+# I/O functions related to Watson Natural Language Processing on the IBM Cloud.
+
+import pandas as pd
+import pyarrow as pa
+
+from text_extensions_for_pandas.array import CharSpanArray, TokenSpanArray
+
+
+def watson_nlp_parse_response(response):
```

Comment:
This function needs a detailed docstring and a type hint for the return value. What is the type of `response`? What does the function return? What expectations does this function have on the contents of `response` beyond its Python type?